library/VTKExtensions/Rendering/Testing/TestF3DOpenGLGridMapper.cxx

```diff
@@ -4,18 +4,66 @@
 #include "vtkF3DOpenGLGridMapper.h"
 
 #include <iostream>
+#include <sstream>
+
+std::string boundsToStr(const double* bounds)
```

Comment:
Functions should start with an upper case letter

Reply:
However, for this, what about using a class `BoundsTest` and with a constructor taking the name and the bounds, and implement `operator==` for comparison and `operator<<` for streaming?

Reply:
`std::array` already has `==` right?
how about
```cpp
template<class T, std::size_t N>
ostream& operator<<(std::ostream& o, const std::array<T, N>& arr)
{
  std::copy(arr.cbegin(), arr.cend(), std::ostream_iterator<T>(o, " "));
  return o;
}

bool CheckBounds(const std::string& name, vtkF3DOpenGLGridMapper* mapper, //
  double x0, double x1, double y0, double y1, double z0, double z1)
{
  std::array<double, 6> actual;
  mapper->GetBounds(actual.data());
  const std::array<double, 6> expected = { x0, x1, y0, y1, z0, z1 };
  if (actual != expected)
  {
    std::cerr << "wrong bounds for " << name << " failed:" << std::endl;
    std::cerr << "  got " << actual << std::endl;
    std::cerr << "  expected " << expected << std::endl;
    return false;
  }
  return true;
}
```?